sources/packages/backend/apps/queue-consumers/src/processors/schedulers/cas-integration/_tests_/cas-send-invoices.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,278 @@
+            approvalStatus: CASInvoiceBatchApprovalStatus.Approved,
+          },
+        },
+      ),
+    );
+    // Creates full-time application with receipts, and invoices details.
+    await saveFakeInvoiceIntoBatchWithInvoiceDetails(
+      db,
+      {
+        casInvoiceBatch,
+        creator: systemUsersService.systemUser,
+      },
+      {
+        offeringIntensity: OfferingIntensity.fullTime,
+        casSupplierInitialValues: {
+          supplierStatus: SupplierStatus.VerifiedManually,
+          supplierNumber: "111111",
+        },
+      },
+    );
+    // Queued job.
+    const mockedJob = mockBullJob<void>();
+
+    // Act
+    const result = await processor.processQueue(mockedJob.job);
+
+    // Assert
+    const casInvoice = await db.casInvoice.findOne({
+      select: {
+        id: true,
+        invoiceNumber: true,
+      },
+      where: {
+        casInvoiceBatch: { id: casInvoiceBatch.id },
+      },
+    });
+    expect(result).toStrictEqual(["Process finalized with success."]);
+    expect(
+      mockedJob.containLogMessages([
+        "Checking for pending invoices.",
+        "Executing 1 pending invoice(s) sent to CAS.",
+        `Processing pending invoice: ${casInvoice.invoiceNumber}.`,
+        `Pending invoice payload invoice number: ${casInvoice.invoiceNumber}.`,
+        "Invoice sent to CAS SUCCEEDED.",
+        "CAS invoice(s) sent job completed.",
+      ]),
+    ).toBe(true);
+  });
+
+  it("Should contain log no pending invoices when there are no pending invoices found.", async () => {
+    // Arrange
+    // Queued job.
+    const mockedJob = mockBullJob<void>();
+
+    // Act
+    const result = await processor.processQueue(mockedJob.job);
+
+    // Assert
+    expect(result).toStrictEqual(["Process finalized with success."]);
+    expect(
+      mockedJob.containLogMessages([
+        "Checking for pending invoices.",
+        "No pending invoices found.",
+        "CAS invoice(s) sent job completed.",
+      ]),
+    ).toBe(true);
+  });
+
+  it("Should capture the error in invoices when sent to CAS and know errors are returned.", async () => {
+    //Arrange
+    // Create invoice batch to generate the report.
+    const casInvoiceBatch = await db.casInvoiceBatch.save(
+      createFakeCASInvoiceBatch(
+        {
+          creator: systemUsersService.systemUser,
+        },
+        {
+          initialValue: {
+            approvalStatus: CASInvoiceBatchApprovalStatus.Approved,
+          },
+        },
+      ),
+    );
+    // Creates full-time application with receipts, and invoices details.
+    await saveFakeInvoiceIntoBatchWithInvoiceDetails(
+      db,
+      {
+        casInvoiceBatch,
+        creator: systemUsersService.systemUser,
+      },
+      {
+        offeringIntensity: OfferingIntensity.fullTime,
+        casSupplierInitialValues: {
+          supplierStatus: SupplierStatus.VerifiedManually,
+          supplierNumber: "111111",
+        },
+      },
+    );
```

Comment:
If the same setup is repeated it can be inside a helper method at the end of this file.